### PR TITLE
Retry button report to memory

### PR DESCRIPTION
**Pull request by cvaldez98**
PR for retry button (#37) and report to memory (#25) combined. 
All of the suggestions and changes were added and I ran npm run prettier just before this.

### CODE REVIEW

File: src/renderer/components/path-chooser.tsx [hunk 1]
```diff
@@ -1,34 +1,23 @@
...
-        appState.absPath = filePaths[0];
-        writeReport(appState.results, filePaths[0], appState.fileName);
-      }
-    }
-  );
+  const filePaths = remote.dialog.showOpenDialog(currentWindow, {
+    title: 'Choose where to add report',
+    properties: ['openDirectory']
+  });
+  if (filePaths != undefined) {
```
Comment (felixrieseberg) [inline src/renderer/components/path-chooser.tsx]:
```suggestion
  if (filePaths) {
```

File: src/renderer/components/results.tsx [hunk 2]
```diff
@@ -1,157 +1,140 @@
...
 
-  return (
-    <Card elevation={Elevation.ONE} className="result-card">
-      {doneElements}
-      <div ref={listRef as any}>{resultElements}</div>
-    </Card>
-  );
-};
+  private renderIndividualResult(suiteResult: SuiteResult): Array<JSX.Element> {
+    // const { testsDone, slackClosed } = this.props;
```
Comment (felixrieseberg) [inline src/renderer/components/results.tsx]: ✂️ Cut this line

File: src/renderer/components/results.tsx [hunk 3]
```diff
@@ -1,157 +1,140 @@
...
-                appState.fileName,
-                appState.absPath,
-                succeeded
-              );
+    function retryTest(
+      testName: string,
+      suiteName: string,
+      testsDone: TestSuite[]
+    ) {
+      testsDone.forEach(testSuite => {
```
Comment (felixrieseberg) [inline src/renderer/components/results.tsx]: 🤔 Didn't we improve this?

File: src/renderer/state.ts [hunk 4]
```diff
@@ -45,6 +45,11 @@ export class AppState {
       .toLocaleTimeString()
       .replace(/:/g, '-')
       .replace(' ', '') + '.txt';
+  @observable public report: string = '';
```
Comment (felixrieseberg) [inline src/renderer/state.ts]:
```suggestion

  @observable public report: string = '';
```

File: src/renderer/components/results.tsx (hunk 3 above)
Comment (cvaldez98) [reply src/renderer/components/results.tsx]: Ohhh I thought we already pushed it so I just left it as is

File: src/renderer/components/results.tsx (hunk 2 above)
Comment (cvaldez98) [reply src/renderer/components/results.tsx]: Forgot to revert sorry!

**Pull request merged**